src/core/lib/event_engine/thread_pool.cc

```diff
@@ -40,15 +40,24 @@ namespace {
 GPR_THREAD_LOCAL(bool) g_threadpool_thread;
 }  // namespace
 
-void ThreadPool::StartThread(StatePtr state) {
+void ThreadPool::StartThread(StatePtr state, bool throttled) {
   state->thread_count.Add();
+  struct ThreadArg {
+    StatePtr state;
+    bool throttled;
+  };
   grpc_core::Thread(
       "event_engine",
       [](void* arg) {
+        std::unique_ptr<ThreadArg> a(static_cast<ThreadArg*>(arg));
         g_threadpool_thread = true;
-        ThreadFunc(*std::unique_ptr<StatePtr>(static_cast<StatePtr*>(arg)));
+        if (a->throttled) {
+          GPR_ASSERT(a->state->starting_thread.exchange(
```

Comment:
Since this atomic exchange would allow for new threads to be created again, should this atomic exchange happen somewhere inside ThreadPool::Queue::Step method when it becomes clear that this particular thread is useful/engaged in processing callbacks ?

Reply:
We'd need to clear it prior to thread exit regardless... and any other location would get a bunch more complicated.
I think instead some delay before exiting if there's nothing to do would be advantageous (but out of scope for this change).